drivers/include/periph/uart.h

```diff
@@ -134,6 +134,23 @@ enum {
  */
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg);
 
+/**
+ * @brief   Recalculate Baudrate for UART device
+ *
+ * The UART device will be initialized with the following configuration:
+ * - 8 data bits
+ * - no parity
+ * - 1 stop bit
+ * - baudrate as given
+ *
+ * @param[in] uart          UART device to initialize
+ * @param[in] baudrate      desired baudrate in baud/s
+ *
+ * @return                  UART_OK on success
+ * @return                  UART_NOBAUD on inapplicable baudrate
+ */
+int uart_set_baudrate(uart_t uart, uint32_t baudrate);
```

Comment:
I think this API introduction would need some discussion, but for now I'd we should call it `uart_calc_baudrate` or something like that. I'm not sure how many platforms support this feature but I'd say to put this function as static in the stm32l code.

Reply:
Since this function is introduced to be used in pm.h, if it's static it wouldn't be available. I could just declare it here and as extern in pm.h, would that work? doesn't seem very clean though.

Reply:
You don't need to add it to the API but only as a local function for the stm32 family let's say. You can put it in cpu/stm32_common/include/periph_cpu_common.h